File: code/pascal/storing-using-data/FtoC.pas
//
// Program: FtoC.pas
// Converts a temperature from Fahrenheit to Celsius
//
program FtoC;
uses SysUtils;
    
    // Convert the tempF parameter to Celsius and return the result
    function ConvertFtoC(tempF: Double): Double;
    begin
        result := (5/9) * (tempF-32);
    end;
    
    // Read a temperature in Fahrenheit, and output the value
    // converted to Celsius
    procedure Main();
    var
        tempF, tempC: Double;
    begin
        Write('Please enter temperature in Fahrenheit: ');
        ReadLn(tempF);
        tempC := ConvertFtoC(tempF);
        WriteLn('This is ', tempC:4:2, ' in Celsius');
    end;

begin
    Main();
end.

File: code/c/storing-using-data/f-to-c.c
// ... header missing

double fahrenheit_to_celsius(double fahrenheit)
{
    return (5.0 / 9.0) * (fahrenheit-32);
}

int main()
{
    double input_temp;
    string line;

    write("Please enter temperature in Fahrenheit: ");
    line = read_line();
    input_temp = convert_to_double(line);

    write_line("This is " + to_string(fahrenheit_to_celsius(input_temp)) + " in Celsius");
    return 0;
}